### PR TITLE
New Feature Flags, including support for video resolution

### PR DESCRIPTION
**Pull request by jimmyjose-dev**
Resolves Reduce video Quality #20 

Added support for:

1) Flag indicating the local and (maximum) remote video resolution. 
    Overrides the server configuration.
    Supports 180p, 360p, 480p, 720p

2) Flag indicating if conference timer should be enabled.

3) Flag indicating if server URL change is enabled.

4) Flag indicating if the video share button should be enabled.

Sample usage for video resolution in dart file

`featureFlags[FeatureFlagEnum.RESOLUTION] = FeatureFlagVideoResolution.LD_RESOLUTION;`

Supported values for resolution
 For video resolution at 180p use LD_RESOLUTION
 For video resolution at 360p use MD_RESOLUTION
 For video resolution at 480p (SD) use SD_RESOLUTION
 For video resolution at 720p (HD) use HD_RESOLUTION

### CODE REVIEW

**Comment by TBG-FR** [comment]
Awesome, thanks for updating with the new feature flags ! Looks good to me, even if I can't try it right now

File: lib/jitsi_meet.dart [hunk 1]
```diff
@@ -229,13 +229,13 @@ class JitsiMeetingOptions
   String userEmail;
   String iosAppBarRGBAColor;
 
-  Map<FeatureFlagEnum, bool> featureFlags = new HashMap();
+  Map<FeatureFlagEnum, dynamic> featureFlags = new HashMap();
```
Comment (tuantvu) [inline lib/jitsi_meet.dart:232]: I understand why we moved to dynamic, but the tighter type check is very nice. Is there some code design you can come up with that makes it hard for clients to put in the wrong type of value for a feature flag (other than having to read the docs)? Should we separate out the integer flags? Should we go away from an enum map and use a class to represent flags?

File: android/src/main/kotlin/com/gunschu/jitsi_meet/JitsiMeetPlugin.kt [hunk 2]
```diff
@@ -138,10 +138,19 @@ public class JitsiMeetPlugin() : FlutterPlugin, MethodCallHandler, ActivityAware
...
+            featureFlags!!.forEach { 
+                (key, value) -> if(value is Boolean ) {
+
+                val boolVal  = value.toString().toBoolean()
+                optionsBuilder.setFeatureFlag(key,boolVal)
+            } else{
+                val intVal  = value.toString().toInt()
+                optionsBuilder.setFeatureFlag(key,intVal)
+            }
+                }
```
Comment (tuantvu) [inline android/src/main/kotlin/com/gunschu/jitsi_meet/JitsiMeetPlugin.kt:153]: Extra bracket or bad formatting?

**Review by tuantvu — CHANGES_REQUESTED** [review]

File: lib/jitsi_meet.dart (hunk 1 above)
Comment (TBG-FR) [reply lib/jitsi_meet.dart:232]: I agree with @tuantvu ! A class could be a good alternative, considering that aspect.

Comment (jimmyjose-dev) [reply lib/jitsi_meet.dart:232]: Sounds good to me, I will try to rewrite it.

Comment (TBG-FR) [reply lib/jitsi_meet.dart:232]: If you're short in time, I can try it, just tell me :wink:

Comment (jimmyjose-dev) [reply lib/jitsi_meet.dart:232]: @TBG-FR 
Sure brother, I am bit occupied with PR for the main Jitsi Repo and some other stuff, if you wanna take stab at it go ahead :) . I can spend time on this over the weekend.

If you do start with the work, do drop a message, so we both don't end up working on the same thing.
More power to you brother :)

File: android/src/main/kotlin/com/gunschu/jitsi_meet/JitsiMeetPlugin.kt (hunk 2 above)
Comment (nilsreichardt) [reply android/src/main/kotlin/com/gunschu/jitsi_meet/JitsiMeetPlugin.kt:153]: Same thought 🤔

File: lib/jitsi_meet.dart (hunk 1 above)
Comment (TBG-FR) [reply lib/jitsi_meet.dart:232]: I didn't had time yet ahah, any news on your side ?

Comment (jimmyjose-dev) [reply lib/jitsi_meet.dart:232]: Nope :( .. don't think I would have time this week as well.

**Comment by jimmyjose-dev** [comment]
This feature has been merged from another pull request... closing this PR.

**Pull request closed without merging**